src/vm/crst.h

```diff
@@ -296,7 +296,11 @@ friend class Crst;
 #endif
 
     union {
+#ifdef CROSSGEN_COMPILE
```

Comment:
Other cross-compilation (debugger, DAC) works correctly when we use `T_CRITICAL_SECTION` here? I cannot be convinced because I'm not familiar with these components.